src/rules/no-trailing-zeros.ts

```diff
@@ -0,0 +1,54 @@
+import type { AST } from "yaml-eslint-parser";
+import { createRule } from "../utils";
+
+export default createRule("no-trailing-zeros", {
+  meta: {
+    docs: {
+      description: "disallow trailing zeros",
+      categories: ["standard"],
+      extensionRule: false,
+      layout: true,
+    },
+    fixable: "code",
+    schema: [],
+    messages: {
+      wrongZeros: "Trailing zeros are not allowed, fix to `{{fixed}}`.",
+    },
+    type: "layout",
+  },
+  create(context) {
+    if (!context.parserServices.isYAML) {
+      return {};
+    }
+
+    return {
+      YAMLScalar(node: AST.YAMLScalar) {
+        if (node.style !== "plain") {
+          return;
+        } else if (typeof node.value !== "number") {
+          return;
+        }
+
+        // https://github.com/stylelint/stylelint/blob/650f597806679a3d2eb57672c931b1a5e2acd0d6/lib/rules/number-no-trailing-zeros/index.js#LL67C25-L67C25
+        const match = /(\d*)\.(\d{0,100}?)(0+)(?:\D|$)/.exec(node.strValue);
```

Comment:
I think the current implementation strictly has false positives. Could you change it to a method that does not use regex?

Reply:
The regex is copied from Stylelint -- it has been time-tested.

I think reimplementing it in a different way would increase complexity while decreasing reliability. Are there any other concerns about it?

Reply:
I don't think writing without regex would be complicated. For example:

```js
if (!node.strValue.endsWith('0')) {
    return
}
const parts = node.strValue.split('.')
if (parts.length !== 2) {
    return
}
while (parts[1].endsWith('0')) {
    parts[1] = parts[1].slice(0, -1)
}
const fixed = parts.join('.')
```

Reply:
Okay, I'll try to integrate it.